contracts/finance/VestingWalletWithCliff.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {SafeCast} from "../utils/math/SafeCast.sol";
+import {VestingWallet} from "./VestingWallet.sol";
+
+/**
+ * @dev Extension of {VestingWallet} that adds a cliff to the vesting schedule.
+ */
+abstract contract VestingWalletWithCliff is VestingWallet {
+    using SafeCast for *;
+
+    uint64 private immutable _cliff;
+
+    error InvalidCliffDuration(uint64 cliffSeconds, uint64 durationSeconds);
+
+    /**
+     * @dev Sets the sender as the initial owner, the beneficiary as the pending owner, the start timestamp, the
+     * vesting duration and the duration of the cliff of the vesting wallet.
+     */
+    constructor(uint64 cliffSeconds) {
```

Comment:
I see `durationSeconds` is used in `VestingWallet`, but there it's straightforward because the parameter is the actual duration. In the case of `cliffSeconds`, it's added over the `start`, which may seem confusing.

We can:

- Rename `cliffSeconds` to `cliffDurationSeconds`
- Rename the `cliff()` function to `cliffTimestamp`

Leaning towards number 1:

```suggestion
    constructor(uint64 cliffDurationSeconds) {
```